hoot-core/src/main/cpp/hoot/core/conflate/polygon/BuildingMatchCreator.cpp

```diff
@@ -128,20 +130,43 @@ class BuildingMatchVisitor : public ConstElementVisitor
           }
           else
           {
-            _result.push_back(m);
+            tempMatches.push_back(m);
             neighborCount++;
           }
         }
       }
     }
 
+    if (_reviewMatchesOtherThanOneToOne && neighborCount > 1)
+    {
+      for (std::vector<Match*>::iterator it = tempMatches.begin(); it != tempMatches.end(); ++it)
+      {
+        Match* match = *it;
+        //Not proud of this, but not sure what else to do at this point w/o having to change the
+        //Match interface.
```

Comment:
Definitely not proud of this const cast.  Couldn't come up with any other way to implement this without changing the Match interface...a big change.  This change is kind of going against the original spirit of the match creator workflow.  So, maybe if we have to make changes like this in other places in the code in the future, we might consider a redesign of the interfaces....or maybe not.